Add facade backend (#53)
iqm-finland/qiskit-on-iqm: docs/user_guide.rst

    job.result().get_counts()

Above, we use an :class:`.IQMFakeAdonis` instance to run a noisy simulation of ``circuit`` on a simulated 5-qubit Adonis
chip. If you want to customize the noise model instead of using the default one provided by :class:`.IQMFakeAdonis`, you
can create a copy of the fake Adonis instance with updated error profile:

.. code-block:: python

    error_profile = backend.error_profile
    error_profile.t1s['QB2'] = 30000.0  # Change T1 time of QB2 as example
    custom_fake_backend = backend.copy_with_error_profile(error_profile)

Running a quantum circuit on a facade backend
---------------------------------------------

Circuits can be executed against a "simulated" environment: an IQM server that has no real quantum computer hardware.
Results from such executions are random bits. This may be useful when developing and testing software integrations.

> Since Qiskit on IQM contains :class:`.IQMFakeAdonis`, it is possible to combine the "simulated" remote execution with

Reviewer: `.IQMFacadeBackend` ?

Author: I meant to refer to `IQMFakeAdonis` that is described in the section above this. I.e., since `IQMFakeAdonis` exists, we also created `IQMFacadeBackend`.

Author: Updated this sentence.

> Qiskit on IQM contains :class:`.IQMFacadeBackend`, which allows to combine the "simulated" remote execution with proper
> local simulation. This way you can get realistic results while not using a real quantum computer.

Reviewer: ```suggestion
Qiskit on IQM contains :class:`.IQMFacadeBackend`, which allows to combine the "simulated" remote execution with a local
noisy quantum circuit simulation. This way you can both validate your integration as well as get an idea of the expected circuit execution results.
```

To run a circuit this way, use the `facade_adonis` backend retrieved from the provider. Note that the provider must be
initialized with the URL of a quantum computer with the equivalent architecture (i.e. names of qubits, their

> connectivity and supported supported gates should match the 5-qubit Adonis architecture).

Reviewer 2: ```suggestion
connectivity and supported gates should match the 5-qubit Adonis architecture).
```

To run a circuit this way, use the `facade_adonis` backend retrieved from the provider. Note that the provider must be
initialized with the URL of a quantum computer with the equivalent architecture (i.e. names of qubits, their
connectivity and supported gates should match the 5-qubit Adonis architecture).

.. code-block:: python

    from qiskit import execute, QuantumCircuit
    from qiskit_iqm import IQMProvider

    circuit = QuantumCircuit(2)
    circuit.h(0)
    circuit.cx(0, 1)
    circuit.measure_all()

    provider = IQMProvider("https://demo.qc.iqm.fi/cocos/")
    backend = provider.get_backend('facade_adonis')
    job = execute(circuit, backend, shots=1000)
    job.result().get_counts()

.. note::

   When a classical register is added to the circuit, Qiskit fills it with classical bits of value 0 by default. If the
   register is not used later, and the circuit is submitted to the IQM server, the results will not contain those
   0-filled bits. To make sure the facade backend returns results in the same format as a real IQM server,
   :meth:`.IQMFacadeBackend.run` checks for the presence of unused classical registers, and fails with an error if there
   are any.

> 
> One or more circuits contain unused classical registers

Reviewer: ```suggestion
```
